sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -368,7 +369,10 @@ abstract class UnixTime extends BinaryExpression with ExpectsInputTypes {
           t.asInstanceOf[Long] / 1000000L
         case StringType if right.foldable =>
           if (constFormat != null) {
-            Try(new SimpleDateFormat(constFormat.toString).parse(
+            if (formatter == null) {
+              formatter = Try(new SimpleDateFormat(constFormat.toString)).getOrElse(null)
```

Comment:
Didn't want to throw the error back as it would break the earlier functionality.  Eearlier it was returning null when any exception (i.e, could be constFormat being null, or parsing error) was thrown.  

Creating the formatter upfront in the recent commit and handling null earlier itself, to have minimal changes.